frontend/src/createStream/__tests__/CreateStreamPageContents.test.jsx

```diff
@@ -0,0 +1,410 @@
+        receiverChannelIndex: -1
+      };
+      const someIndex = 88;
+      const expectedState = {
+        receiverChannelIndex: someIndex
+      };
+
+      wrapper.setState(startingState);
+      wrapper.instance().setReceiverChannelIndex(someIndex);
+
+      expect(wrapper.state()).toMatchObject(expectedState);
+    });
+  });
+  describe("createStream() function", () => {
+    const device = (num) =>
+      new DeviceInfo("", "", "", "", `hello${num}`, "Online", [
+        new InputChannelInfo("", "", num)
+      ]);
+    const senders = [device(0), device(1), device(2)];
+    const receivers = [device(3), device(4), device(5)];
+
+    beforeEach(() => {
+      jest.spyOn(StreamApi, "createStream");
+      wrapper = Enzyme.shallow(<CreateStreamPageContents />);
+    });
+
+    describe("when reciver and sender are valid devices", () => {
+      describe("when receiverChannel and senderChannel are valid channels", () => {
+        it("calls StreamApi.createStream with expected arguments", () => {
+          wrapper.setState({
+            senders,
+            receivers,
+            senderDeviceIndex: 1, // device(1)
+            senderChannelIndex: 0,
+            receiverDeviceIndex: 2, // device(4)
+            receiverChannelIndex: 0
+          });
+          wrapper.instance().createStream();
+          const expected = {
+            receiver: {
+              chId: receivers[2].channels[0].id
+            },
+            sender: {
+              chId: senders[1].channels[0].id
+            }
+          };
+
+          expect(StreamApi.createStream).toBeCalledWith(
+            expected.receiver.chId,
+            expected.sender.chId
+          );
+        });
+        it("when StreamApi.createStream resolves, calls snackbar() wiht expected args", async () => {
+          wrapper.setState({
+            senders,
+            receivers,
+            senderDeviceIndex: 1, // device(1)
+            senderChannelIndex: 0,
+            receiverDeviceIndex: 2, // device(4)
+            receiverChannelIndex: 0
+          });
+          jest.spyOn(SnackbarHandler, "snackbar").mockImplementation();
+          jest
+            .spyOn(StreamApi, "createStream")
+            .mockImplementation(() => Promise.resolve());
+          wrapper.instance().createStream();
+
+          await flushPromises();
+
+          const expected = ["success", "Successfully created stream!"];
+          expect(SnackbarHandler.snackbar).toBeCalledWith(
+            expected[0],
+            expected[1]
+          );
+        });
+        it("when StreamApi.createStream rejects, calls snackbar() with expected args", async () => {
+          wrapper.setState({
+            senders,
+            receivers,
+            senderDeviceIndex: 1, // device(1)
+            senderChannelIndex: 0,
+            receiverDeviceIndex: 2, // device(4)
+            receiverChannelIndex: 0
+          });
+          jest.spyOn(SnackbarHandler, "snackbar").mockImplementation();
```

Comment:
Just something I wanted to point out before I jump into a review later on. I noticed this when I was getting updates from my email, but this isn't how you spy on the snackbar function, mocking it actually doesn't really do anything. If you look at PR #422, you can have a look at tests that actually correctly spy on snackbar being called.

In short, you would simply have to assign the jest.spyOn to a constant, like this: 
`const snackbarSpy = jest.spyOn(SnackbarHandler, "snackbar");`

and in your tests you would do something like this:
`expect(snackbarSpy).toHaveBeenCalledTimes(1);`
`expect(snackbarSpy).toHaveBeenCalledWith("success", "Successfully created stream!");`

Reply:
I don't want it to actually call the snackbar, since it requires the <SnackBarMessage/> component's handleOpen to exist. Instead I just want to check that It's being called. Which is what I'm doing. That's why the function has a mock implementation